Kitodo/src/main/java/org/kitodo/production/security/SecurityConfig.java

```diff
@@ -99,99 +99,182 @@ protected void configure(HttpSecurity http) throws Exception {
         http.sessionManagement().maximumSessions(1).sessionRegistry(getSessionRegistry())
                 .expiredUrl(LOGIN_PAGE);
 
-        // viewAll... Authority to view list of entities
-        // view...... Authority to view entity at edit page
-        // edit...... Authority to change and save entities at edit page
-        http.authorizeRequests()
-            .antMatchers("/pages/clientEdit.jsf*").hasAnyAuthority(
-                EDIT_CLIENT + GLOBAL,
-                EDIT_CLIENT + CLIENT_ANY,
-                VIEW_CLIENT + GLOBAL,
-                VIEW_CLIENT + CLIENT_ANY)
+        // site specific rules
+        authorizeSpecificPages(http);
 
-            .antMatchers("/pages/indexingPage.jsf").hasAnyAuthority(
-                "editIndex_" + GLOBAL,
-                "viewIndex_" + GLOBAL)
+        // more general rules should be at end
```

Comment:
Just out of curiosity: why should general rules be at the end?

Reply:
See the examples in https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/config/annotation/web/builders/HttpSecurity.html#authorizeHttpRequests(org.springframework.security.config.Customizer) and there  `Note that the matchers are considered in order. Therefore, the following is invalid because the first matcher matches every request and will never get to the second mapping`